src/ORM/FieldType/DBForeignKey.php

```diff
@@ -70,11 +58,11 @@ public function scaffoldFormField($title = null, $params = null)
         return $field;
     }
 
-    public function setValue($value, $record = null, $markChanged = true)
+    public function setValue(mixed $value, null|array|ViewableData $record = null, bool $markChanged = true): static
```

Comment:
Keeping this as `ViewableData` mostly for consistency, but also because it's conceivable to save a reference to a `DataObject` into some arbitrary non-database-backed model.